src/Si7021_temp_humidity_sensor.c

```diff
@@ -0,0 +1,68 @@
+#include "stm32f0xx.h"
+#include "FreeRTOS.h"
+#include "task.h"
+#include "I2C.h"
+#include "semphr.h"
+
+//Time out
+#define I2C_TIMEOUT          2000
+
+//Si7021 register commands
+#define CMD_MEASURE_RH_HM            0xE5
+#define CMD_MEASURE_RH_NM            0xF5
+#define CMD_MEASURE_TEMP_HM          0xE3
+#define CMD_MEASURE_TEMP_NH          0xF3
+#define CMD_READ_TEMP_PREV_RH        0xE0
+#define CMD_RESET                    0xFE
+
+
+//Si7021 I2C slave addresses
+#define AD_READ         0b1000000
+#define AD_WRITE        0b1000000
+
+
+//Take the relative humidity and return it as a percentage
+extern uint16_t Update_Humidity() {
+
+	uint8_t humidityAddress = CMD_MEASURE_RH_HM;
+
+	xSemaphoreTake(I2C_mutex, I2C_TIMEOUT);
+
+	I2C_write(AD_WRITE, 1, &humidityAddress);
```

Comment:
All the I2C commands should be put inside an if statement and should only execute if the mutex was successfully taken. Refer this the documentation for more information. https://www.freertos.org/a00122.html